### PR TITLE
Avoid recreate instance when cluster not ready

### PR DESCRIPTION
**Pull request by ron-gal**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigtable: avoided re-creation of instance when cluster is still updating and storage type changed
```

```release-note:enhancement
bigtable: added `state` output attribute to `google_bigtable_instance` clusters
```

### CODE REVIEW

**Comment by rileykarson** [comment]
I see this is marked as draft- let me know when you'd like me to make a review pass.

File: mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go [hunk 1]
```diff
@@ -63,6 +63,10 @@ func ResourceBigtableInstance() *schema.Resource {
 				Description: `A block of cluster configuration options. This can be specified at least once.`,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"state": {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go:66]: We'll want to add a `Description` to this field for wrapper tools (i.e. KCC) and a website entry under the Attributes reference, the path of the field being `cluster.0.state`: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/bigtable_instance.html.markdown#attributes-reference

File: mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go [hunk 2]
```diff
@@ -624,11 +642,6 @@ func resourceBigtableInstanceClusterReorderTypeList(_ context.Context, diff *sch
 		}
 	}
 
-	err := diff.SetNew("cluster", orderedClusters)
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go:627]: I'm not entirely certain that this is the cause of the failures, but I think that `diff.SetNew` modified the internal state of the resource, affecting future `diff` method calls, and we're seeing different behaviour now that's caused a handful of tests to fail. We may need to break up the function to end here, starting a new function afterwards, and combine them. (We could also stub out the behaviour of the ResourceDiff object, but I am not confident we'd accurately recreate it in a stub)

Comment (ron-gal) [reply mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go:627]: Fixed

File: mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go (hunk 1 above)
Comment (ron-gal) [reply mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go]: Fixed

**Review by rileykarson — APPROVED** [review]

**Review by rileykarson — COMMENTED** [review]
Ah, wait, one last thing- since we've added the new `state` output we'll still need to document it in the website: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/bigtable_instance.html.markdown?rgh-link-date=2023-09-20T20%3A02%3A22Z#attributes-reference

**Pull request merged**